### ISSUE TITLE
[BUG] When padding a timeseries the column name is removed.

### ISSUE THREAD

**Issue by kdekker-private** [sktime/sktime#6679]
**Describe the bug**
When padding a timeseries the column name is removed.

**To Reproduce**

```python
from sktime.datasets import load_airline
from sktime.datatypes import convert_to
from sktime.transformations.panel.padder import PaddingTransformer

y = load_airline()
original_name = y.name

padder = PaddingTransformer()
y_new = padder.fit_transform(y)
new_name = y_new.name

original_name == new_name
```

**Expected behavior**
Original name should be the same as the new name.

**Versions**
0.30.1

**Comment by fkiraly** [comment]
This is an interesting one.

There are two causes:

* inside the estimator, columns are dropped
* a long conversions round trip.

For this we have to understand what happens internally in transformers if they receive an input the internal logic is not written for:

1. first the boilerplate checks whether the scitype of what is passed equals the inner scitype. In this case, the transformer is written for `Panel` data, i.e., collections of time series. The data passed is a single time series, so of `Series` scitype.
2. If there is a mismatch, we upcast if the inner type has more hierarchy levels, otherwise we broadcast, i.e., apply the transformer to all instances. For upcast, broadcast, and downcast, we need to convert into a subset of mtypes for which this is implemented.
3. If there is a match between outer and inner scitype, we now check mtpes, i.e., machine type representations.
4. If the inner type matches the outer type, we just pass the data on. If it does not, we do a round trip of conversion to and back conversion from.
5. if in step 2 we had a mismatch, we now need to compose if we broadcast, and downcast if we upcast. If we converted in 2, we need to back-convert.

The error sneaks in through the fact that data types are not 1:1 convertible, as for instance `pandas` containers have fields like `index` and `name`, while other types such as `numpy` or `dask` based ones do not. So we memorize, in step 4, in `_converter_store_X` these pieces of information, to restore them at the end of the round trip.

However, this does not happen for step 2. In the case here, we downcast a panel data set to `pd.DataFrame`, and then convert it to `pd.Series`. Due to how the logic currently is, we do not know whether the single column label at that stage arose from the `name` of an initial `pd.Series`, or is just a placeholder. So it is removed and does not end up in `name`, in the back-conversion step in 5.

There are a couple of ways how we could remedy that, but they are all somewhat substantial:

* option 1: ensure that the upcast/downcast cycle has memory of things like `name`
* option 2: we rewrite the padder that internally it uses flat `pandas` based types instead of `nested_univ`. This fixes it for the padder only.
* option 3: we use that placeholder column is always the column "0". So if we encounter a column of another name, we do not overwrite with `None` in `convert_MvS_to_UvS_as_Series`. This will remove the bug except in the very specific case where the user had specified a `name` initially, passed a `pd.Series` to an estimator that does the exact round trip as above (convert, upcast, convert, backconvert, downcast, backconvert), and the `name` specified is exactly the integer (not string) `0`.

Thoughts, @kdekker-private?

Also, out of interest, why are you applying the padder to a single series? Does this arise in some kind of edge case for your application?

**Comment by kdekker-private** [comment]
Option 1: Feels like the cleanest.
I think it is sufficient. 
It does not only resolve the issue for padding, but given your explanation it seems also for more transformers.
It does not resolve the case in which the output has a different number of columns compared to the input.
However, in the latter case I would expect that the transformer has extra logic to cover for this.

Option 2: Feels like you will have to re-apply it too often as most transformers keep a constant column length.

Option 3: I would not go for this solution as transformers are often chained in a sequence. This option will imply that for all transformers that come later the column name is incorrect for the pandas dataframe input/output.
